internal/sync/syncer.go

```diff
@@ -233,6 +240,22 @@ func (s *Syncer) runOnce(ctx context.Context, key string, worker int) error {
 	if err != nil {
 		return fmt.Errorf("failed to get the manifest of key %q, %w", key, err)
 	}
+
+	namespacedName := types.NamespacedName{
+		Namespace: gitSync.Namespace,
+		Name:      gitSync.Name,
+	}
+	lastSyncedCommitHash, err := getCommitStatus(ctx, s.client, namespacedName, &numaLogger)
+	if err != nil {
+		return fmt.Errorf("failed to get the current commit status of key %q, %w", key, err)
```

Comment:
One thing to note is that we may generally only hit this error (if we do at all) because we've been rate limited by the K8S API Server. We could consider deciding that the only reason we were checking this to begin with was as an optimization, so we could theoretically continue. But I'm okay if maybe we want to just keep this as is.

Reply:
Yeah, if we hit an error because of rate limiting, we will continue for the next round of syncing. So let's keep it simple and stick to the current logic.

Reply:
sounds good